### PR TITLE
Add cluster pause webhook

### LINKED ISSUE #1850: Package upgrade on cluster version change
This is a follow up to https://github.com/vmware-tanzu/tanzu-framework/pull/1713. Packages managed by ClusterBootstrap must be updated with new versions and configs when status.resolvedTKR is different from TKR label on a cluster. 

1. CLI/other clients will set paused to true when setting a new version in KCP
2. TKR controller will label the cluster with version corresponding to KCP.
3. ClusterBootstrap controller must roll out package upgrades with the following requirements

   a. Ensure packages have node selector with new TKR. (Assume TKR controller has already labeled     nodes with TKR version
   b. Ensure deployment strategy is RollingUpdate
   c. Ensure daemonset strategy is OnDelete. In prev investigation we found kapp-controller gets stuck on daemonset. Investigate and log upstream issue if needed.

### LINKED ISSUE #1971: Add cluster pause webhook to addons manager package
(This is used to request new product features)

**Describe the feature request**
Add cluster pause webhook to addons manager package

Cluster pause webhook is added in https://github.com/vmware-tanzu/tanzu-framework/pull/1965

**Describe alternatives you've considered**

**Affected product area (please put an X in all that apply)**

- [ ] APIs
- [x] Addons
- [ ] CLI
- [ ] Docs
- [ ] IAM
- [ ] Installation
- [ ] Plugin
- [ ] Security
- [ ] Test and Release
- [ ] User Experience

**Additional context**

### PR DESCRIPTION
**Pull request by blc1996**
Signed-off-by: Lucheng Bao <luchengb@vmware.com>

### What this PR does / why we need it
Add cluster pause webhook

This webhook will set the cluster to pause state and add TKG pause annotation when topology.version is bumped.
Later, ClusterBootstrap controller will unset the pause after it rolls out the package upgrades.

### Which issue(s) this PR fixes

Fixes #1850 

### Describe testing done for PR
Added testcase in env test

### Release note

```release-note
package-based-lcm: Add cluster pause webhook
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

File: pkg/v1/webhooks/cluster_pause_webhook.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+// Default satisfies the defaulting webhook interface.
+func (wh *Cluster) Default(ctx context.Context, obj runtime.Object) error {
+	cluster, ok := obj.(*clusterv1.Cluster)
+	if !ok {
+		return apierrors.NewBadRequest(fmt.Sprintf("expected a Cluster but got a %T", obj))
+	}
+
+	// Try to get the current cluster CR so we can compare the version
+	oldCluster := &clusterv1.Cluster{}
```
Comment (vijaykatam) [inline pkg/v1/webhooks/cluster_pause_webhook.go:45]: nit: currentCluster

Comment (vijaykatam) [inline pkg/v1/webhooks/cluster_pause_webhook.go:45]: nit: currentCluster

**Review by vijaykatam — COMMENTED** [review]
Minor comments. Change looks good. Please log an issue to followup and integrate into addons/main.go

File: pkg/v1/webhooks/cluster_pause_webhook.go (hunk 1 above)
Comment (blc1996) [reply pkg/v1/webhooks/cluster_pause_webhook.go]: Updated

Comment (blc1996) [reply pkg/v1/webhooks/cluster_pause_webhook.go]: Updated

**Comment by blc1996** [comment]
> Minor comments. Change looks good. Please log an issue to followup and integrate into addons/main.go

^^ Issue logged https://github.com/vmware-tanzu/tanzu-framework/issues/1971

**Review by vijaykatam — APPROVED** [review]
LGTM

**Pull request merged**